addons/website_sale/static/src/js/website_sale_recently_viewed.js

```diff
@@ -56,31 +56,32 @@ publicWidget.registry.productsRecentlyViewedSnippet = publicWidget.Widget.extend
             return;
         }
         var products = res['products'];
-        if (products && products.length) {
-            var mobileProducts = [], webProducts = [], productsTemp = [];
-            _.each(products, function (product) {
-                if (productsTemp.length === 4) {
-                    webProducts.push(productsTemp);
-                    productsTemp = [];
-                }
-                productsTemp.push(product);
-                mobileProducts.push([product]);
-            });
-            if (productsTemp.length) {
+        var mobileProducts = [], webProducts = [], productsTemp = [];
+        _.each(products, function (product) {
+            if (productsTemp.length === 4) {
                 webProducts.push(productsTemp);
+                productsTemp = [];
             }
+            productsTemp.push(product);
+            mobileProducts.push([product]);
+        });
+        if (productsTemp.length) {
+            webProducts.push(productsTemp);
+        }
 
-            this.mobileCarousel = $(qweb.render('website_sale.productsRecentlyViewed', {
-                uniqueId: this.uniqueId,
-                productFrame: 1,
-                productsGroups: mobileProducts,
-            }));
-            this.webCarousel = $(qweb.render('website_sale.productsRecentlyViewed', {
-                uniqueId: this.uniqueId,
-                productFrame: 4,
-                productsGroups: webProducts,
-            }));
-            this._addCarousel();
+        this.mobileCarousel = $(qweb.render('website_sale.productsRecentlyViewed', {
+            uniqueId: this.uniqueId,
+            productFrame: 1,
+            productsGroups: mobileProducts,
+        }));
+        this.webCarousel = $(qweb.render('website_sale.productsRecentlyViewed', {
+            uniqueId: this.uniqueId,
+            productFrame: 4,
+            productsGroups: webProducts,
+        }));
+        this._addCarousel();
+        if (!products || !products.length) {
+            this.$el.removeClass('pt24 pb24').animate({'min-height': '', 'height': ''}, {duration: 1000});
```

Comment:
Removing the padding classes is useless I guess, just need overflow hidden.

Reply:
Overflow is nice but the padding stays :/